src/reqs.c

```diff
@@ -681,7 +681,11 @@ static int get_all_headers (int fd, hashmap_t hashofheaders)
                 /*
                  * Append the new line to the current header field.
                  */
-                tmp = (char *) saferealloc (header, len + linelen);
+                if( NULL == header ) {
```

Comment:
what's the point of this check ?

according to `man 3p realloc`

> The  realloc() function shall deallocate the old object pointed to by ptr and return a pointer to a new object that has the size specified by size.
> ...
> If  ptr  is a null pointer, realloc() shall be equivalent to malloc() for the specified size.

Reply:
I found this in 1.8.4, it use safemalloc.... this bug is already fixed in the mater